scalatest/src/main/scala/org/scalatest/tools/SuiteDiscoveryHelper.scala

```diff
@@ -294,13 +295,13 @@ private[scalatest] object SuiteDiscoveryHelper {
       if (!dir.isDirectory)
         throw new IllegalArgumentException
 
-      val subDirs = for (entry <- dir.listFiles.toList; if entry.isDirectory) yield entry
+      val subDirs = for (entry <- dir.listFiles.toList.map(_.nn); if entry.isDirectory) yield entry
       val fileLists: List[List[String]] = 
         for (subDir <- subDirs) 
           yield listFilesInDir(subDir, prependPrevName(prevName, subDir.getName))
 
       val files: List[String] =
-        for (entry <- dir.listFiles.toList; if !entry.isDirectory)
+        for (entry <- dir.listFiles.toList.map(_.nn); if !entry.isDirectory)
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.